pkg/catalog/endpoint.go

```diff
@@ -18,3 +20,39 @@ func (mc *MeshCatalog) ListEndpointsForService(svc service.MeshService) ([]endpo
 	}
 	return endpoints, nil
 }
+
+// GetResolvableServiceEndpoints returns the resolvable set of endpoint over which a service is accessible using its FQDN
+func (mc *MeshCatalog) GetResolvableServiceEndpoints(svc service.MeshService) ([]endpoint.Endpoint, error) {
+	// TODO: Move the implmentation of this function to be provider-specific. Currently, the providers might
```

Comment:
I would raise an issue for this and reference it in the comment.

Reply:
https://github.com/openservicemesh/osm/issues/1655